Verse/src/DecoderDescriptors/Tree/TreeDecoderStream.cs

```diff
@@ -28,9 +28,20 @@ public void Dispose()
 
 		public bool TryDecode(out TEntity entity)
 		{
-			entity = this.constructor();
+			var entityValue = this.constructor();
 
-			return this.callback(this.reader, this.state, ref entity);
+			var result = this.callback(this.reader, this.state, ref entityValue, out var isVoid);
+
+			if (isVoid)

```

Comment:
Please simplify to `entity = isVoid ? default : entityValue` (but you maybe don't need this statement at all, cf. above comment).

Reply:
done